src/server.rs

```diff
@@ -208,11 +205,60 @@ impl LspServer {
         options: LspServerOptions,
     ) -> Self {
         Self {
-            client: Arc::new(Mutex::new(client)),
+            client,
             store: Arc::new(Mutex::new(HashMap::new())),
             options,
         }
     }
+    // If we have a client back to the editor report any diagnostics found in the document
+    async fn publish_diagnostics(&self, key: &lsp::Url, text: &str) {
+        if let Some(client) = &self.client {
+            match flux::new_semantic_analyzer(
+                flux::semantic::AnalyzerConfig::default(),
```

Comment:
The analyzer doesn't hold state so its not important other than the config, but this code path explicitly has a different config than other uses of the analyzer because we want it to report errors instead of ignore them. 

Most of the time the LSP parses/analyzes it just wants the semantic graph and doesn't want the errors. Because of our use of the Result type we get only one or the other not both. So this analyzer has a different config to actually capture and report errors.